File: c++/leetcode/1446. Consecutive Characters.cpp
class Solution
{
public:
  int maxPower(string s)
  {
    int power = 1;

    const int sz = s.size() - 1;

    for (int i = 0; i < sz; ++i)
    {
      int count = 1;

      while (i < sz && s[i + 1] == s[i] && ++i)
        power = max(power, ++count);
    }

    return power;
  }
};

File: javascript/leetcode/1446. Consecutive Characters.js
/**
 * @param {string} s
 * @return {number}
 */
var maxPower = function (s) {
  let power = 1;

  const sz = s.length - 1;

  for (let i = 0; i < sz; ++i) {
    let count = 1;

    while (i < sz && s[i + 1] === s[i] && ++i) power = Math.max(power, ++count);
  }

  return power;
};
